cpp/include/raft/handle.hpp

```diff
@@ -126,26 +142,42 @@ class handle_t {
     return cusparse_handle_;
   }
 
-  cudaStream_t get_internal_stream(int sid) const { return streams_[sid]; }
-  int get_num_internal_streams() const { return num_streams_; }
+  // legacy compatibility for cuML
+  cudaStream_t get_internal_stream(int sid) const {
+    return streams_.get_stream(sid).value();
+  }
+  // new accessor return rmm::cuda_stream_view
+  rmm::cuda_stream_view get_internal_stream_view(int sid) const {
+    return streams_.get_stream(sid);
+  }
+
+  int get_num_internal_streams() const { return streams_.get_pool_size(); }
   std::vector<cudaStream_t> get_internal_streams() const {
     std::vector<cudaStream_t> int_streams_vec;
-    for (auto s : streams_) {
-      int_streams_vec.push_back(s);
+    for (int i = 0; i < get_num_internal_streams(); i++) {
+      int_streams_vec.push_back(get_internal_stream(i));
     }
     return int_streams_vec;
   }
 
+  handle_t get_handle_from_internal_pool(
+    int stream_id, int n_streams = kNumDefaultWorkerStreams) const {
```

Comment:
What happens if kNumDefaultWorkerStreams > 0 (say 10)?

The parent handle will have a stream pool of 10 and the child handle will get its default stream from the parent but will have an additional stream pool with 10 streams?

Reply:
Yes. This is for the case where a prim is given a light handle but need to make concurrent calls to other prims.